### PR TITLE
x265: 2.9 -> 3.0

### PR DESCRIPTION
**Pull request by jlesquembre**
###### Motivation for this change
Update to 3.0

Also, since version 2.6, x265 it's using `nasm` instead of `yasm`, see https://bitbucket.org/multicoreware/x265/src/feec4bdf98663ac4a2561b63e1ccec423c7d26a4/build/README.txt#lines-37
That means that the x265 version from nix, wasn't using any of the CPU optimizations

I also compiled with NUMA support by default. Not 100% sure, but it seems that NUMA systems will get a big performance win, while for non NUMA systems this change should be un-noticeable. For more info see:
https://bitbucket.org/multicoreware/x265/commits/62b8fe990df5e560834e9b567a913238d8dd398e
https://bitbucket.org/multicoreware/x265/issues/108/numa-pools

I compressed a short video (~1 minute) with the same options, and the performance change was huge, it went down from 22 to 8 minutes

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by jlesquembre** [comment]
@c0bw3b I hope it's ok if I ping you. Since you merged the last x265 update, and recently merged a very similar PR #61683, it would be nice if you can review my PR.

The upgrade to 3.0 and `nasm` are easy to review, I can split the NUMA support in a different PR, if it helps, to at least get the CPU optimization for x265 back

File: pkgs/development/libraries/x265/default.nix [hunk 1]
```diff
@@ -67,7 +54,9 @@ stdenv.mkDerivation rec {
     rm $out/lib/*.a
   '';
 
-  nativeBuildInputs = [ cmake yasm ];
+  nativeBuildInputs = [ cmake nasm
+  ] ++ stdenv.lib.optional numaSupport numactl;
```
Comment (c0bw3b) [inline pkgs/development/libraries/x265/default.nix]: (nitpick)
Slightly unusual style. I guess you can keep all of it on a single line

**Review by c0bw3b — COMMENTED** [review]
I'll try and build _some_ reverse deps to check for potential regressions.
Although it's a major release, the changelog doesn't seem to announce breaking changes:
https://x265.readthedocs.io/en/latest/releasenotes.html#version-3-0

**Comment by jlesquembre** [comment]
Ups, sorry, I mess up with a git rebase.
I'm closing this PR and opening a new one

**Pull request closed without merging**